### PR TITLE
Tests for not displaying uid(WIP)

### LINKED ISSUE #2676: PLANNING ISSUE: Multi party authentication
This is the checklist for the OAuth Login 
# Implementation common to all providers
* [x] Add omniauth gem #2381
* [x] Add fiagro gem #2531
* [x] Install Openssl #2745 
* [x] Make localhost work under https connection #2745
* [x] Write Documentaion for openssl #2745 
* [x] Write Documentation for omniauth and related gems #2855 #2848
* [x] Define routes #2531
* [x] Modify UserTag to acts as identiy model #2737 #2639
* [x] Add OmniAuth Capability to User Model #2771
* [x] Link provider to a user #2771
* [x] Delete provider from a user #2771
* [x] User_controller update #2771
* [x] User_session_controller update  - #2771
* [x] Wiki Page https://publiclab.org/oauth

# Google Provider
* [x] Add omniauth-google-oauth2 gem #2771
* [x] Add OmniAuth Configuration to initializer #2771 
* [x] OAuth testing hash #2771 
* [x] Write user session controller tests for the login through Google #2771 
* [x] Write integration tests for the login through Google #2771 
* [x] Set up developers app (discussion in #2771)
* [x] Docker file changes #2771 
* [x] header  #2866 
* [x] /profile/edit page  #2850
* [x] login page  #2851 
* [x] sign up page  #2852 

# Github Provider
* [x] Add omniauth-github gem #2686
* [x] Add OmniAuth Configuration to initializer #2873
* [x] Set up developers app #2856 
* [x] Docker file changes #2875
* [x] OAuth testing hash #2874
* [x] Write user session controller tests for the login through Github  #2915 
* [x] Write integration tests for the login through Github #2914
* [x] header  #2975 
* [x] /profile/edit page #2975 
* [x] login page  #2975 
* [x] sign up page  #2975 

# Twitter Provider
* [x] Add omniauth-twitter gem #2686 
* [x] Set up developers app #2893
* [x] Docker file changes #2947 
* [x] Add OmniAuth Configuration to initializer #2948
* [x] OAuth testing hash #2961
* [x] Write user sessions controller tests for the login through Twitter #2962
* [x] Write integration tests for the login through Twitter #2963
* [x] header  #2986
* [x] /profile/edit page #2986
* [x] login page #2986
* [x] sign up page  #2986

# Facebook Provider
* [x] Add omniauth-facebook gem #2381 
* [x] Set up developers app #2867
* [x] Docker file changes #2960
* [x] Add OmniAuth Configuration to initializer#2959
* [x] OAuth testing hash #2965
* [x] Write user session controller tests for the login through Facebook #2966
* [x] Write integration tests for the login through Facebook #2964
* [x] header  #2987
* [x] /profile/edit page#2987
* [x] login page #2987
* [x] sign up page#2987

# Handling Security  Vulnerabilities
* [x] Random Number for password field #3031 #3071
* [x] Introduction of Password checker field into user model #3032
* [x] Uid field filter on profile page - only the admin and the user himself can see the usertag #3038
* [x] Uid field filter on profile page - only the admin and the user himself can see the usertag tests #3139
* [x] Trimming of the uid from the usertag when showed on profile page #3066
* [x] Tests for above #3139

# Front End
* [x]  header #2866
* [x] Login Page #3011
* [x] Sign Up Page #3012
* [x] Edit Page #3014

### PR DESCRIPTION
**Pull request by SidharthBansal**
Closes #2676

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [ ] code is in uniquely-named feature branch and has no merge conflicts
* [ ] PR is descriptively titled
* [ ] PR body includes `fixes #0000`-style reference to original issue #
* [ ] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
This is under progress.
@jywarren I am having fever. So cant focus much. Sorrry for the delay in prs. I will try to send prs as soon as I can.

File: test/functional/users_controller_test.rb [hunk 1]
```diff
@@ -257,4 +257,57 @@ def setup
...
+    UserSession.create(users(:jeff))
+    user = users(:jeff)
+    auth = { "provider" => "github", "uid" => "123456789"}
+    uid = user.id
+    identity = UserTag.create_with_omniauth(auth, uid)
+    get :profile
+    assert_response :redirect
+    assert_redirected_to '/profile/jeff'
+    s = "span#tag_"+identity.id.to_s
+    assert_select s,'omniauth:github'
```
Comment (SidharthBansal) [inline test/functional/users_controller_test.rb]: @jywarren can you please tell me the what I am doing wrong here. I checked in the view it is https://github.com/publiclab/plots2/blob/de4d09faf10d16ab2f15291373ca954ebaba60ee/app/views/tag/_tags.html.erb#L20-L25
It says span with id as tag_+tag.id. Is tag.id not same as usertag.id?

Comment (jywarren) [reply test/functional/users_controller_test.rb]: Hmm, can you check it locally? Maybe also look for the total count of items like `span.label` to be sure they're appearing at all?

Comment (SidharthBansal) [reply test/functional/users_controller_test.rb]: Yeah I am trying to debug this. I am trying my level best to write these test asap. It is giving errors.

On localhost everything is working fine. Also in production things are working fine. Just trying to debug the tests.

**Comment by jywarren** [comment]
ok, no worries, and thanks for your hard work!
Maybe @publiclab/reviewers could help if you can't figure it out?

**Comment by SidharthBansal** [comment]
I am closing this and turning these tests into gci help wanted issues.
Thanks

**Comment by SidharthBansal** [comment]
The system has changed considerably from the time I pushed these changes. So there is a need to close this one.
Thanks

**Pull request closed without merging**